polar-core/src/bindings.rs

```diff
@@ -157,12 +157,16 @@ impl BindingManager {
     ///
     /// If the binding succeeds, the new expression is returned as a goal. Otherwise,
     /// an error is returned.
-    fn partial_bind(&mut self, partial: Operation, var: &Symbol, val: Term) -> PolarResult<Goal> {
+    fn partial_bind(
+        &mut self,
+        partial: Operation,
+        var: &Symbol,
+        val: Term,
+    ) -> Result<Goal, RuntimeError> {
```

Comment:
Nit: is there any benefit to seeing the runtime error here? I think most files only expose one error type, so perhaps `type Result<T> = Result<T, RuntimeError>` etc. at the top of the file?

Reply:
5bdc9985